batchflow/models/tf/encoder_decoder.py

```diff
@@ -233,14 +258,20 @@ def encoder(cls, inputs, name='encoder', **kwargs):
 
                 for i in range(steps):
                     with tf.variable_scope('encoder-'+str(i)):
-                        # Preprocess tensor with given block
-                        args = {**kwargs, **block_args, **unpack_args(block_args, i, steps)} # enforce priority of keys
-                        x = base_block(x, name='pre', **args)
-
-                        # Downsampling
-                        if downsample.get('layout') is not None:
-                            args = {**kwargs, **downsample, **unpack_args(downsample, i, steps)}
-                            x = conv_block(x, name='downsample-{}'.format(i), **args)
+                        # Make all the args
+                        args = {**kwargs, **block_args, **unpack_args(block_args, i, steps)}
+                        downsample_args = {**kwargs, **downsample, **unpack_args(downsample, i, steps)}
+
+                        if order in ['bd', 'bp']: # block -> downsample
+                            x = base_block(x, name='block', **args)
+                            if downsample.get('layout') is not None:
+                                x = conv_block(x, name='downsample', **downsample_args)
+                        elif order in ['db', 'pb']: # downsample -> block
+                            if downsample.get('layout') is not None:
+                                x = conv_block(x, name='downsample', **downsample_args)
+                            x = base_block(x, name='block', **args)
+                        else:
+                            raise ValueError('Unknown order {}, use one of ("bd", "db")'.format(order))
```

Comment:
I would check order format and then
```
for comp in order:
    if comp is 'b':
        do block
    else:
       do downsample
```
to avoid code copy-paste

Reply:
That is definitely better. That would also allow to use multiple `blocks` in a row, though I don't know whether we want it.